fdbserver/BlobManifest.actor.cpp

```diff
@@ -149,76 +358,57 @@ class BlobManifestDumper : public ReferenceCounted<BlobManifestDumper> {
 					state KeySelectorRef begin = firstGreaterOrEqual(range.begin);
 					state KeySelectorRef end = firstGreaterOrEqual(range.end);
 					loop {
-						RangeResult result = wait(tr.getRange(begin, end, limits, Snapshot::True));
+						state RangeResult result = wait(tr.getRange(begin, end, limits, Snapshot::True));
 						for (auto& row : result) {
-							rows.push_back_deep(rows.arena(), KeyValueRef(row.key, row.value));
+							wait(splitter->append(splitter, KeyValueRef(row.key, row.value)));
 						}
 						if (!result.more) {
 							break;
 						}
 						if (result.readThrough.present()) {
 							begin = firstGreaterOrEqual(result.readThrough.get());
 						} else {
-							begin = firstGreaterThan(result.end()[-1].key);
+							begin = firstGreaterThan(result.back().key);
 						}
 					}
 				}
-				return rows;
+
+				// store the read vesion
+				Version readVersion = wait(tr.getReadVersion());
```

Comment:
well noted. Currently we could read all manifest rows within(much less than) 5 sec based my testing with 500GB fdb. we'll need to rethink the design if it takes longer time. We'll have more insights when testing with >1TB fdb cluster. we can discuss this topic more after. 

Regarding error handling, BlobManifestFileSplitter.reset() will reset the spitter and re-write the manifest files from the beginning. The other option is to simply rethrow the error and wait for next scheduled dump as you mentioned. This one seems simpler.  I would like to do more testing with a large fdb cluster and see how much overhead it may introduce.. then decide with option we should go.

Reply:
Makes sense. I think that reset should also delete the files that were already written or in flight by the previous iteration?

Reply:
I think there are potentially some weird issues here with relying on file name idempotency.
For example, say that between retries of the manifest, a bunch of granules were merged and the number of manifest files required went down from 8 to 6, and the first time around we had written manifest files 1-7 before retrying.
This would leak file 7 as well as potentially creating an inconsistency or problems if you tried to include the old manifest file 7 in the new set that was supposed to be 6.
It also used to be the case that s3 overwrites were not strongly consistent, and i'm not sure about other cloud's blob stores, so we might not want to rely on that anyway. I think it'd be preferable to do at least one of deleting files between rounds and/or using a unique filename for each round.

Also, is the mainfest filename format described anywhere? I must have missed it

Reply:
the manifest file name would be "<epoch>.<seq>.<segment>.manifest". see fileName(). 

Ok. it seems not good to rely on s3 consistencies for file overwrites. added file deletion in reset(). I'm still thinking eventually we may not need to reset at all. we can simply stop dumping and let next dumper generates a good manifest. let's see how frequently this error may happen with a large cluster testing.